src/dist.rs

```diff
@@ -43,10 +43,30 @@ pub(crate) mod rust {
     for (slice_org, slice_ref) in
       plane_org.rows_iter().take(h).zip(plane_ref.rows_iter())
     {
-      sum += slice_org
+      let mut iter_org = slice_org[..w].chunks_exact(4);
+      let mut iter_ref = slice_ref[..w].chunks_exact(4);
+      for (chunk_org, chunk_ref) in iter_org.by_ref().zip(iter_ref.by_ref()) {
+        let chunk_org = wide::i32x4::from([
+          i32::cast_from(chunk_org[0]),
+          i32::cast_from(chunk_org[1]),
+          i32::cast_from(chunk_org[2]),
+          i32::cast_from(chunk_org[3]),
+        ]);
+        let chunk_ref = wide::i32x4::from([
+          i32::cast_from(chunk_ref[0]),
+          i32::cast_from(chunk_ref[1]),
+          i32::cast_from(chunk_ref[2]),
+          i32::cast_from(chunk_ref[3]),
+        ]);
+        let sad = (chunk_org - chunk_ref).abs();
+        for i in sad.to_array() {
+          sum += i as u32;
+        }
```

Comment:
Alternatively, without the `wide` crate:
```suggestion
        let chunk_org = [
          i32::cast_from(chunk_org[0]),
          i32::cast_from(chunk_org[1]),
          i32::cast_from(chunk_org[2]),
          i32::cast_from(chunk_org[3]),
        ];
        let chunk_ref = [
          i32::cast_from(chunk_ref[0]),
          i32::cast_from(chunk_ref[1]),
          i32::cast_from(chunk_ref[2]),
          i32::cast_from(chunk_ref[3]),
        ];
        sum += chunk_org
          .iter()
          .zip(chunk_ref)
          .map(|(a, b)| (a - b).abs() as u32)
          .sum::<u32>();
```

Reply:
Interestingly the compiler seems to be failing to accomplish this--at least given that I have `target-cpu=native` enabled, the `wide` crate is vectorizing with intrinsics and the generated asm is showing `vmovd` and `vpinsrd` instructions. The plain Rust code seems to be failing to autovectorize. (I think without `target-cpu=native` the compiler still defaults to sse2 which should still allow wide to use some intrinsics.)

Reply:
For both 8-bit and 10-bit encodes, `get_sad` only accounts for 0.03% of CPU time after this change.
With the above suggestion, perf shows that the loop is auto-vectorized and hot:
```
  0.51 │285:   xor          %r14d,%r14d
  1.19 │       nop
  4.50 │290:   vmovd        (%rdi,%r14,1),%xmm0
 15.55 │       vmovd        (%r10,%r14,1),%xmm1
 14.23 │       vpsadbw      %xmm0,%xmm1,%xmm0
 15.73 │       vmovd        %xmm0,%ebx
 12.56 │       add          %ebx,%eax
  2.83 │       add          $0x4,%r14
  0.52 │       cmp          %r14,%rdx
  1.77 │     ↑ jne          290
```

Reply:
Interesting. I was using `cargo asm` to show the asm and it was not showing autovectorization.